### PR TITLE
Updated "Creating a Bot Account" Docs

### PR DESCRIPTION
**Pull request by ToasterUwU**
## Summary

This updates the "Creating a Bot Account" Docs to be in sync with the actual website from Discord that recently changed to hide tokens permanently for security reasons. It also tells people to tick the "applications.commands" box if they are using slash, member, or message commands.

This is my go-to guide when I need someone to make a bot account and the last time I used it for that (a few days ago) I had to give some notes with it to not make any mistakes.

PS: Sorry for these two other commits, they are from the main branch of my fork which I used for another Pull Request. But I forgot to make another branch for that, so I had to fix this in this one.

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [X] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: docs/discord.rst [hunk 1]
```diff
@@ -35,7 +35,7 @@ Creating a Bot account is a pretty straightforward process.
     .. image:: /images/discord_bot_user_options.png
         :alt: How the Bot User options should look like for most people.
 
-7. Copy the token using the "Copy" button.
+7. Generate the token using the "Reset Token" button. You might have to confirm and enter your current 2FA Code.
```
Comment (MaskDuck) [inline docs/discord.rst:38]:
```suggestion
7. Generate the token using the "Reset Token" button. You have to confirm and enter your current 2FA Code, even if you don't have 2FA enabled.
```

Comment (ToasterUwU) [reply docs/discord.rst:38]: Is 2FA actually required and also promoted even if not enabled?

Comment (MaskDuck) [reply docs/discord.rst:38]: wait nvm im wrongc

Comment (ToasterUwU) [reply docs/discord.rst:38]: > wait nvm im wrongc

OK, everything else fine?

File: docs/discord.rst [hunk 2]
```diff
@@ -71,10 +76,10 @@ If you want to invite your bot you must create an invite URL for it.
     .. image:: /images/discord_oauth2.png
         :alt: How the OAuth2 page should look like.
 
-5. Tick the "bot" checkbox under "scopes".
+5. Tick the "bot" checkbox under "scopes". If you make use of Application Commands (Slash, Message and Member Commands) also tick "applications.commands".
```
[the remark this answers is not in the document]
Comment (ToasterUwU) [reply docs/discord.rst]: Changed

File: docs/discord.rst (hunk 1 above)
Comment (ooliver1) [reply docs/discord.rst:38]: can confirm its required

**Review by ooliver1 — APPROVED** [review]
awesome

**Review by ooliver1 — APPROVED** [review]

**Review by RamzziSudip — APPROVED** [review]

**Review by koala9712 — APPROVED** [review]

**Pull request merged**